components/browser/icons/src/main/java/mozilla/components/browser/icons/compose/IconLoaderState.kt

```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.browser.icons.compose
+
+import androidx.compose.ui.graphics.painter.Painter
+import mozilla.components.browser.icons.BrowserIcons
+import mozilla.components.browser.icons.Icon.Source
+
+/**
+ * The state an [IconLoaderScope] is in.
+ */
+sealed class IconLoaderState {
```

Comment:
Can a `IconLoader` be in `Failed` state?

Reply:
No, if no icon can be loaded then `BrowserIcons` generates an icon for that website. If the app wants to do something else in this case then it can still check `icon.source` and see if it is set to `GENERATOR`.